lib/workers/repository/package-files.ts

```diff
@@ -109,8 +109,16 @@ export class PackageFiles {
           for (const dep of packageFile.deps) {
             const ver = dep.currentValue;
             const digest = dep.currentDigest;
-            const version =
-              ver && digest ? `${ver}@${digest}` : `${digest ?? ver!}`;
+            const lock = dep.lockedVersion;
+            let version;
+            if (ver || digest) {
+              version =
+                ver && digest ? `${ver}@${digest}` : `${digest ?? ver ?? lock}`;
```

Comment:
```suggestion
                ver && digest ? `${ver}@${digest}` : `${digest ?? ver}`;
```
lock makes no sense here